src/hangar/cli/cli.py

```diff
@@ -535,136 +536,172 @@ def server(overwrite, ip, port, timeout):
+                extension = os.path.splitext(fpath)[1].strip('.')
+                loaded = external.load(fpath, plugin=plugin, extension=extension, **kwargs)
+                if isinstance(loaded, types.GeneratorType):
+                    for arr, fname in loaded:
+                        if not overwrite and fname in aset:
+                            continue
+                        try:
+                            aset[fname] = arr
+                        except ValueError as e:
+                            click.echo(e)
+                else:
+                    arr, fname = loaded
+                    if not overwrite and fname in aset:
+                        continue
+                    try:
+                        aset[fname] = arr
+                    except ValueError as e:
+                        click.echo(e)
     finally:
         co.close()
 
 
-@main.command(name='export')
-@click.argument('startpoint', nargs=1, required=True)
+@main.command(name='export', context_settings=dict(allow_extra_args=True, ignore_unknown_options=True,))
 @click.argument('arrayset', required=True)
-@click.option('-o', '--out', required=True, help='Path to export the data to.')
-@click.option('-s', '--sample', default=False, help='Sample name to export')
-@click.option('-f', '--format', 'format_', required=False, help='File format used for exporting.')
+@click.argument('startpoint', nargs=1, default=None, required=False)
+@click.option('-o', '--out', 'outdir', required=False, default=os.getcwd(), help="Directory to export data")
+@click.option('-s', '--sample', default=None, help='Sample name to export. You can optionally specify the type'
+                                                   ' of sample name after the colon (ex. `54:str` or `54:int`)')
+@click.option('-f', '--format', 'format_', required=False, help='File format of output file')
 @click.option('--plugin', required=False, help='override auto-inferred plugin')
 @pass_repo
-def export_data(repo: Repository, startpoint, arrayset, out, sample, format_, plugin):
-    """export ARRAYSET sample data as it existed a STARTPOINT to some format and path.
+@click.pass_context
+def export_data(ctx, repo: Repository, arrayset, outdir, startpoint, sample, format_, plugin):
+    """Export ARRAYSET sample data as it existed a STARTPOINT to some format and path.
+
+    Specifying which sample to be exported is possible by using the switch
+    `--sample` (without this, all the samples in the given arrayset will be
+    exported). Since hangar supports both int and str datatype for the sample
+    name, specifying that while mentioning the sample name might be necessary
+    at times. It is possible to do that by separating the name and type by a
+    colon.
+
+    Example:
+
+       1. if the sample name is string of numeric 10 - ``10:str`` or ``10``
+
+       2. if the sample name is ``sample1`` - ``sample1:str1`` or ``sample1``
+
+       3. if the sample name is an int, let say 10 - ``10:int``
```

Comment:
@lantiga, do you have any suggestions here?

this is the best interface we could come up with, since apparently every other "simple" identifier for a number or a string seems to have some significance to a bash shell before it gets parsed by python.  

Our preferred were: 

- `"3"` to identify a `string 3` (`"` and `'` are escaped) 
- `#3` for an `int 3`(`#` starts a comment)
- ``` `3` ``` for an `string 3` (``` ` ``` enclosed value is directly executed as a command)

Reply:
IMO: If it parses as an int, it's an int. 

If it starts with `str:` or `int:` then it's the thing indicated.

Otherwise, it's a string.

Reply:
Note that this means that all `int:` does is prevent fallback to string in the case where it doesn't parse cleanly. So `int:3` works, and `int:foo` causes the entire program to bomb out.

Reply:
Also, this is bikeshedding, but I much prefer the prefix form to the suffix.

Also, if it's possible for the value to have a colon in it, then I would require the `str:` prefix.

Reply:
Thanks for the feedback! I prefer the prefix `int:5` as well. I'm still hoping there might be something we can use which is a bit "shorter" than prepenting every integer with `int:`, but we may be out of options.